fastdds_python/test/api/test_publisher.py

```diff
@@ -491,3 +491,59 @@ def create_publisher():
            participant.delete_publisher(publisher))
     assert(fastdds.RETCODE_OK ==
            factory.delete_participant(participant))
+
+
+def test_get_datawriter_qos_from_xml():
+
+    with open("test_xml_profile.xml", "r", encoding="utf-8") as file:
+        xml_content = file.read()
+
+    factory = fastdds.DomainParticipantFactory.get_instance()
+    participant = factory.create_participant(
+            0, fastdds.PARTICIPANT_QOS_DEFAULT)
+    publisher = participant.create_publisher(fastdds.PUBLISHER_QOS_DEFAULT)
+
+    qos = fastdds.DataWriterQos()
+    ret = publisher.get_datawriter_qos_from_xml(
+            xml_content, qos, 'test_publisher_profile')
+    assert(fastdds.RETCODE_OK == ret)
+
+    qos_no_name = fastdds.DataWriterQos()
+    ret = publisher.get_datawriter_qos_from_xml(
+            xml_content, qos_no_name)
+    assert(fastdds.RETCODE_OK == ret)
+
+    assert(qos == qos_no_name)
+
+    assert(fastdds.RETCODE_OK ==
+           participant.delete_publisher(publisher))
+    assert(fastdds.RETCODE_OK ==
+           factory.delete_participant(participant))
+
+
+def test_get_default_datawriter_qos_from_xml():
+
+    with open("test_xml_profile.xml", "r", encoding="utf-8") as file:
+        xml_content = file.read()
+
+    factory = fastdds.DomainParticipantFactory.get_instance()
+    participant = factory.create_participant(
+            0, fastdds.PARTICIPANT_QOS_DEFAULT)
+    publisher = participant.create_publisher(fastdds.PUBLISHER_QOS_DEFAULT)
+
+    default_qos = fastdds.DataWriterQos()
+    ret = publisher.get_default_datawriter_qos_from_xml(
+            xml_content, default_qos)
+    assert(fastdds.RETCODE_OK == ret)
+
+    qos = fastdds.DataWriterQos()
+    ret = publisher.get_datawriter_qos_from_xml(
+            xml_content, qos, 'test_publisher_profile')
+    assert(fastdds.RETCODE_OK == ret)
+
+    assert(default_qos == qos)
+
+    assert(fastdds.RETCODE_OK ==
+           participant.delete_publisher(publisher))
+    assert(fastdds.RETCODE_OK ==
+           factory.delete_participant(participant))
```

Comment:
I would use the `delete_contained_entities` method, and explain here why we expect these to match
```suggestion
    # Default participant has the same profile name (only participant defined here)
    assert(default_qos == qos)
    assert(fastdds.RETCODE_OK ==
           participant.delete_contained_entities())
    assert(fastdds.RETCODE_OK ==
           factory.delete_participant(participant))
```